src/mixins/itext_behavior.mixin.js

```diff
@@ -508,7 +509,11 @@
       this._currentCursorOpacity = 0;
 
       this.fire('editing:exited');
-      this.canvas && this.canvas.fire('text:editing:exited', { target: this });
+      isTextChanged && this.fire('modified');
+      if (this.canvas) {
+        this.canvas.fire('text:editing:exited', { target: this });
+        isTextChanged && this.canvas.fire('object:modified', { target: this });
```

Comment:
As any other event, it fires from the object and from the canvas. dev has the choice to register his listener wherever he prefers.